### PR TITLE
JWT support implementation

### PR DESCRIPTION
**Pull request by ValentynSharshon**

### CODE REVIEW

File: src/main/java/com/gmail/woosay333/onlinebookstore/controller/AuthController.java [hunk 1]
```diff
@@ -22,8 +25,10 @@
         description = "Endpoints for registration and login operations.")
 public class AuthController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
-    @PostMapping("/registration")
+    @PostMapping(value = "/registration",
+            produces = "application/json; charset=UTF-8")
```
Comment (vovasalyha) [inline src/main/java/com/gmail/woosay333/onlinebookstore/controller/AuthController.java]: JSON is a default response type, so it's not required.

File: src/main/java/com/gmail/woosay333/onlinebookstore/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+public record UserLoginRequestDto(
+        @NotBlank(message = "The value of the Email field cannot be null or empty")
+        @Email(regexp = "^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,3}$",
+                flags = Pattern.Flag.CASE_INSENSITIVE,
+                message = "The value of the Email field does not match the pattern")
+        @Schema(example = "example@email.com")
+        String email,
+        @NotBlank(message = "The value of the Password field cannot be null or empty")
+        @Length(min = 5, max = 20,
```
Comment (vovasalyha) [inline src/main/java/com/gmail/woosay333/onlinebookstore/dto/user/UserLoginRequestDto.java]: I'd suggest not mixing `jakarta.validation` and `hibernate.validator` annotations. Better use `Size`.

File: src/main/java/com/gmail/woosay333/onlinebookstore/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    public boolean validateToken(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .verifyWith(secret)
+                    .build()
+                    .parseSignedClaims(token);
+            return !claimsJws.getPayload().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new JwtException("Expired or invalid JWT token");
```
Comment (vovasalyha) [inline src/main/java/com/gmail/woosay333/onlinebookstore/security/JwtUtil.java]:
```suggestion
            return false;
```

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -15,3 +15,6 @@ spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
 spring.jpa.properties.hibernate.format_sql=true
 spring.jpa.properties.hibernate.highlight_sql=true
+
+jwt.expiration=300000
+jwt.secret=my-32-character-ultra-secure-and-ultra-long-secret
```
Comment (vovasalyha) [inline src/main/resources/application.properties]: It's better to get sensitive properties from system properties and not pushing it to git.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**